src/IRrecv.cpp

```diff
@@ -56,6 +56,10 @@ static ETSTimer timer;
 }  // namespace _IRrecv
 #endif  // ESP8266
 #if defined(ESP32)
+#if ( defined(ESP_ARDUINO_VERSION) && \
+    (ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(3, 0, 0)) )
+#define _ESP32_ARDUINOV3
+#endif  // Version > 3
```

Comment:
The cruelest of nitpicks (so feel free to ignore :-)
`_ESP32_ARDUINOV3` is a poor/technically incorrect name. It's really a "V3 and/or higher" definition looking at the code.

Perhaps flip the logic and use/create `_ESP32_ARDUINO_PRE_V3` (obviously changing the use of it elsewhere in the code.

Or keep it as is, and simply rename to `_ESP32_ARDUINO_POST_V3_0_0`  ??

Either one is fine IMHO.

Reply:
We have `_ESP32_IRRECV_TIMER_HACK`
It would probably also be wise to consider V4 etc. as well.
So yes `_ESP32_ARDUINO_CORE_V3PLUS` and rename `_ESP32_IRRECV_TIMER_HACK` to `_ESP32_ARDUINO_CORE_V2PLUS` ?

But for now also consider rename being done if/when moving to separate file for this logic